Highway.h

```diff
@@ -2,15 +2,23 @@
 
 #include <vector>
 #include <algorithm>
+#include "Vehicle.h"
+#include "HighwayPatrol.h"
+
+struct Vehicle;
+struct HighwayPatrol;
```

Comment:
Don't forward declare and include.  pick one.   
You're not using either of these types in this header file, so **don't include what you don't use.**
- [ ] revise